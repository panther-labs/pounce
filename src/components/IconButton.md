```jsx harmony
import Arrow from 'icons/arrow.svg';

<IconButton>
  <Arrow />
</IconButton>;
```
