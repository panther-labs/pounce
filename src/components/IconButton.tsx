import React from 'react';
import { css } from '@emotion/core';
import { convertHexToRgba } from 'utils/helpers';
import BaseButton, { BaseButtonProps } from 'components/BaseButton';
import useTheme from 'utils/useTheme';

export interface IconButtonProps extends BaseButtonProps {
  /** Whether the component should appear as "activated". Helpful for menus */
  active?: boolean;

  /** The color variant of the IconButton */
  variant: 'primary' | 'default';
}

/** A wrapper that makes an <a href="/#/Icon">Icon<a> component be clickable */
export const IconButton: React.FC<IconButtonProps> = ({ active, variant, ...rest }) => {
  const { colors } = useTheme();

  return (
    <BaseButton
      {...rest}
      bg="transparent"
      p={3}
      borderRadius="circle"
      css={css`
        color: ${active
          ? colors[variant === 'primary' ? 'primary300' : 'grey400']
          : colors.grey400};

        background-color: ${active &&
          convertHexToRgba(colors[variant === 'primary' ? 'primary300' : 'black'], 0.2)};

        &:hover:not(:active) {
          background-color: ${!active &&
            convertHexToRgba(colors[variant === 'primary' ? 'primary300' : 'black'], 0.1)};
        }

        &:active {
          background-color: ${convertHexToRgba(
            colors[variant === 'primary' ? 'primary300' : 'black'],
            0.2
          )};
        }
      `}
    />
  );
};

IconButton.defaultProps = {
  active: false,
};

export default IconButton;
