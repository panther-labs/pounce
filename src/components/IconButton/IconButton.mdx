An Iconbutton can be solid and have different sizes:

```typescript jsx
<Box>
  <SimpleGrid columns={3} gap={4} mb={2}>
    <Box>
      <IconButton size="small" icon="user" aria-label="User" />
    </Box>
    <Box>
      <IconButton size="small" icon="user" variantColor="teal-500" aria-label="User" />
    </Box>
    <Box>
      <IconButton
        size="small"
        icon="user"
        variantColor="violet-400"
        variantBorderStyle="circle"
        aria-label="User"
      />
    </Box>
  </SimpleGrid>
  <SimpleGrid columns={3} gap={4} mb={2}>
    <Box>
      <IconButton size="medium" icon="user" aria-label="User" />
    </Box>
    <Box>
      <IconButton size="medium" icon="user" variantColor="teal-500" aria-label="User" />
    </Box>
    <Box>
      <IconButton
        size="medium"
        icon="user"
        variantColor="violet-400"
        variantBorderStyle="circle"
        aria-label="User"
      />
    </Box>
  </SimpleGrid>
  <SimpleGrid columns={3} gap={4}>
    <Box>
      <IconButton icon="user" aria-label="User" />
    </Box>
    <Box>
      <IconButton icon="user" variantColor="teal-500" aria-label="User" />
    </Box>
    <Box>
      <IconButton
        icon="user"
        variantColor="violet-400"
        variantBorderStyle="circle"
        aria-label="User"
      />
    </Box>
  </SimpleGrid>
</Box>
```

An IconButton can be `ghost`:

```typescript jsx
<SimpleGrid columns={4} gap={4}>
  <IconButton variant="ghost" icon="user" aria-label="User" variantColor="navyblue-300" />
  <IconButton
    variant="ghost"
    icon="user"
    aria-label="User"
    variantColor="teal-500"
    variantBorderStyle="circle"
  />
  <IconButton variant="ghost" icon="user" aria-label="User" variantColor="navyblue-300" active />
  <IconButton
    variant="ghost"
    icon="user"
    aria-label="User"
    variantColor="teal-500"
    variantBorderStyle="circle"
    active
  />
</SimpleGrid>
```

An IconButton can be `outline`:

```typescript jsx
<SimpleGrid columns={4} gap={4}>
  <IconButton variant="outline" icon="user" aria-label="User" variantColor="navyblue-300" />
  <IconButton
    variant="outline"
    icon="user"
    aria-label="User"
    variantColor="teal-500"
    variantBorderStyle="circle"
  />
  <IconButton variant="outline" icon="user" aria-label="User" variantColor="navyblue-300" active />
  <IconButton
    variant="outline"
    icon="user"
    aria-label="User"
    variantColor="teal-500"
    variantBorderStyle="circle"
    active
  />
</SimpleGrid>
```

An IconButton can have no styling:

```typescript jsx
<SimpleGrid columns={4} gap={4}>
  <IconButton variant="unstyled" icon="user" aria-label="User" />
  <IconButton variant="unstyled" icon="user" aria-label="User" variantBorderStyle="circle" />
</SimpleGrid>
```

An IconButton can have a colored icon:

```typescript jsx
<SimpleGrid columns={4} gap={4}>
  <IconButton icon="user" variantColor="navyblue-300" iconColor="teal-500" aria-label="User" />
  <IconButton
    variant="ghost"
    icon="user"
    variantColor="navyblue-300"
    iconColor="teal-500"
    aria-label="User"
  />
  <IconButton
    variant="outline"
    icon="user"
    variantColor="navyblue-300"
    iconColor="teal-500"
    aria-label="User"
  />
  <IconButton variant="unstyled" icon="user" aria-label="User" iconColor="teal-500" />
</SimpleGrid>
```
