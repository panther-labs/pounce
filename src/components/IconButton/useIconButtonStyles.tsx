import React from 'react';
import { addOpacity, lightenDarkenColor } from '../../utils/helpers';
import { AbstractButtonProps } from '../AbstractButton';
import useTheme from '../../utils/useTheme';
import { IconButtonProps } from './IconButton';
import { Theme } from '../../theme';

type ThemeColor = keyof Theme['colors'];
type UseIconButtonStyles = Required<
  Pick<IconButtonProps, 'variantColor' | 'variant' | 'size' | 'variantBorderStyle'>
>;

export const getUnstyledButtonStyles = (theme: Theme) => {
  return {
    _focus: {
      backgroundColor: addOpacity(theme.colors.white, 0.1),
    },
  };
};

const getOutlineButtonStyles = (theme: Theme, variantColor: ThemeColor) => {
  const themeColor = theme.colors[variantColor];

  const hoverColor = lightenDarkenColor(themeColor, 8);
  const activeColor = lightenDarkenColor(themeColor, -25);
  const focusBorderColor = lightenDarkenColor(themeColor, 85);

  return {
    transition: 'border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms, background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms', // prettier-ignore
    border: '1px solid',
    borderColor: themeColor,
    backgroundColor: 'transparent',
    _hover: {
      backgroundColor: hoverColor,
      borderColor: hoverColor,
    },
    _focus: {
      backgroundColor: hoverColor,
      borderColor: focusBorderColor,
    },
    _active: {
      backgroundColor: activeColor,
      borderColor: activeColor,
    },
  };
};

export const getGhostButtonStyles = (theme: Theme, variantColor: ThemeColor) => {
  const themeColor = theme.colors[variantColor];

  return {
    transition: 'background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms',
    border: '1px solid',
    borderColor: 'transparent',
    backgroundColor: 'transparent',
    _hover: {
      backgroundColor: addOpacity(themeColor, 0.3),
    },
    _focus: {
      backgroundColor: addOpacity(themeColor, 0.3),
    },
    _active: {
      backgroundColor: themeColor,
    },
  };
};

export const getSolidButtonStyles = (theme: Theme, variantColor: ThemeColor) => {
  const themeColor = theme.colors[variantColor];

  const hoverColor = lightenDarkenColor(themeColor, 10);
  const activeColor = lightenDarkenColor(themeColor, -10);
  const focusBorderColor = lightenDarkenColor(themeColor, 85);

  return {
    transition: 'border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms, background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms', // prettier-ignore
    border: '1px solid',
    borderColor: themeColor,
    backgroundColor: themeColor,
    _hover: {
      backgroundColor: hoverColor,
      borderColor: hoverColor,
    },
    _focus: {
      backgroundColor: hoverColor,
      borderColor: focusBorderColor,
    },
    _active: {
      backgroundColor: activeColor,
      borderColor: activeColor,
    },
  };
};

const getSizeButtonStyles = (size: IconButtonProps['size']): AbstractButtonProps => {
  switch (size) {
    case 'small':
      return { height: 24, width: 24 };
    case 'medium':
      return { height: 32, width: 32 };
    case 'large':
    default:
      return { height: 46, width: 46 };
  }
};

const getBorderStyles = (
  variantBorderStyle: IconButtonProps['variantBorderStyle']
): AbstractButtonProps => {
  switch (variantBorderStyle) {
    case 'circle':
      return { borderRadius: 'circle' };
    case 'square':
    default: {
      return { borderRadius: 'medium' };
    }
  }
};

const useIconButtonStyles = ({
  variantColor,
  variantBorderStyle,
  variant,
  size,
}: UseIconButtonStyles): AbstractButtonProps => {
  const theme = useTheme();

  const variantStyles = React.useMemo(() => {
    switch (variant) {
      case 'ghost':
        return getGhostButtonStyles(theme, variantColor);
      case 'unstyled':
        return getUnstyledButtonStyles(theme);
      case 'outline':
        return getOutlineButtonStyles(theme, variantColor);
      case 'solid':
      default:
        return getSolidButtonStyles(theme, variantColor);
    }
  }, [variantColor, variant]);

  const sizeStyles = getSizeButtonStyles(size);
  const borderStyles = getBorderStyles(variantBorderStyle);

  return {
    outline: 'none',
    lineHeight: 'none',
    ...borderStyles,
    ...sizeStyles,
    ...variantStyles,
  } as AbstractButtonProps;
};

export default useIconButtonStyles;
