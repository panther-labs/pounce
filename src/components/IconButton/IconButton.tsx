import React from 'react';
import Icon, { IconProps } from '../Icon';
import useIconButtonStyles from './useIconButtonStyles';
import { BoxProps, NativeAttributes } from '../Box';
import AbstractButton from '../AbstractButton';
import { Theme } from '../../theme';

export interface IconButtonProps extends NativeAttributes<'button'>, Pick<BoxProps, 'as' | 'to'> {
  /** The size of the icon button */
  size?: 'small' | 'medium' | 'large';

  /** The text associated with the icon button */
  'aria-label': string;

  /** The icon present on the button  */
  icon: IconProps['type'];

  /** The style of the icon button */
  variant?: 'solid' | 'ghost' | 'outline' | 'unstyled';

  /** The color scheme of the button */
  variantColor?: keyof Theme['colors'];

  /** The border style of the button  */
  variantBorderStyle?: 'square' | 'circle';

  /** Whether the button is disabled */
  disabled?: boolean;

  /** Whether the button should have a loading spinner next to it */
  loading?: boolean;

  /** Whether the button should always be marked as active. Helpful for menus. */
  active?: boolean;
}

/** A wrapper that makes an <a href="/#/Icon">Icon<a> component be clickable */
export const IconButton = React.forwardRef<HTMLButtonElement, IconButtonProps>(function IconButton(
  {
    variantColor = 'blue-300',
    variantBorderStyle = 'square',
    active = false,
    variant = 'solid',
    size = 'large',
    icon,
    ...rest
  },
  ref
) {
  const styles = useIconButtonStyles({ variantColor, variant, variantBorderStyle, size });

  return (
    <AbstractButton
      aria-pressed={active !== undefined ? active : undefined}
      data-active={active || undefined}
      {...styles}
      {...rest}
      ref={ref}
    >
      <Icon type={icon} size={size} />
    </AbstractButton>
  );
});

export default IconButton;
