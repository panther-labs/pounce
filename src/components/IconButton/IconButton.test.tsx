import React from 'react';
import { renderWithTheme } from 'test-utils';
import IconButton from './index';

describe('IconButton', () => {
  it('renders `small` size correctly', () => {
    const { container } = renderWithTheme(<IconButton size="small" icon="add" aria-label="Test" />);

    expect(container).toMatchSnapshot();
  });

  it('renders `medium` size correctly', () => {
    const { container } = renderWithTheme(
      <IconButton size="medium" icon="add" aria-label="Test" />
    );

    expect(container).toMatchSnapshot();
  });

  it('renders `large` size correctly', () => {
    const { container } = renderWithTheme(<IconButton size="large" icon="add" aria-label="Test" />);

    expect(container).toMatchSnapshot();
  });

  it('renders `variantColor`  and `variants` correctly', () => {
    const { container } = renderWithTheme(
      <>
        <IconButton
          variantColor="teal-500"
          variant="solid"
          size="large"
          icon="add"
          aria-label="Test"
        />
        <IconButton variantColor="pink-700" variant="ghost" icon="brackets" aria-label="Test red" />
        <IconButton variantColor="navyblue-700" icon="timer" aria-label="Test timer" />
      </>
    );

    expect(container).toMatchSnapshot();
  });

  it('renders `square` variantBorderStyle correctly', () => {
    const { container } = renderWithTheme(
      <IconButton variantBorderStyle="square" icon="add" aria-label="Test" />
    );

    expect(container).toMatchSnapshot();
  });

  it('renders `circle` variantBorderStyle correctly', () => {
    const { container } = renderWithTheme(
      <IconButton variantBorderStyle="circle" icon="add" aria-label="Test" />
    );

    expect(container).toMatchSnapshot();
  });
});
