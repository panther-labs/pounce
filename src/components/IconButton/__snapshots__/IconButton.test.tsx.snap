// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders \`circle\` variantBorderStyle correctly 1`] = `
.pounce-0 {
  display: inline-block;
  vertical-align: sub;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  color: currentColor;
}

.pounce-1 {
  outline: none;
  line-height: 1;
  border-radius: 99999px;
  height: 46px;
  width: 46px;
  -webkit-transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border: 1px solid;
  border-color: #0081C5;
  background-color: #0081C5;
  cursor: pointer;
  color: #F6F6F6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.pounce-1:hover {
  background-color: #0a8bcf;
  border-color: #0a8bcf;
}

.pounce-1:focus {
  background-color: #0a8bcf;
  border-color: #55d6ff;
}

.pounce-1:active,
.pounce-1[data-active=true] {
  background-color: #0077bb;
  border-color: #0077bb;
}

<div>
  <button
    aria-label="Test"
    aria-pressed="false"
    class="pounce-1"
    type="button"
  >
    <svg
      class="pounce-0"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 13.1429H13.1429V20H10.8571V13.1429H4V10.8571H10.8571V4H13.1429V10.8571H20V13.1429Z"
      />
    </svg>
  </button>
</div>
`;

exports[`IconButton renders \`large\` size correctly 1`] = `
.pounce-1 {
  outline: none;
  line-height: 1;
  border-radius: 4px;
  height: 46px;
  width: 46px;
  -webkit-transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border: 1px solid;
  border-color: #0081C5;
  background-color: #0081C5;
  cursor: pointer;
  color: #F6F6F6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.pounce-1:hover {
  background-color: #0a8bcf;
  border-color: #0a8bcf;
}

.pounce-1:focus {
  background-color: #0a8bcf;
  border-color: #55d6ff;
}

.pounce-1:active,
.pounce-1[data-active=true] {
  background-color: #0077bb;
  border-color: #0077bb;
}

.pounce-0 {
  display: inline-block;
  vertical-align: sub;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  color: currentColor;
}

<div>
  <button
    aria-label="Test"
    aria-pressed="false"
    class="pounce-1"
    type="button"
  >
    <svg
      class="pounce-0"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 13.1429H13.1429V20H10.8571V13.1429H4V10.8571H10.8571V4H13.1429V10.8571H20V13.1429Z"
      />
    </svg>
  </button>
</div>
`;

exports[`IconButton renders \`medium\` size correctly 1`] = `
.pounce-1 {
  outline: none;
  line-height: 1;
  border-radius: 4px;
  height: 32px;
  width: 32px;
  -webkit-transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border: 1px solid;
  border-color: #0081C5;
  background-color: #0081C5;
  cursor: pointer;
  color: #F6F6F6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.pounce-1:hover {
  background-color: #0a8bcf;
  border-color: #0a8bcf;
}

.pounce-1:focus {
  background-color: #0a8bcf;
  border-color: #55d6ff;
}

.pounce-1:active,
.pounce-1[data-active=true] {
  background-color: #0077bb;
  border-color: #0077bb;
}

.pounce-0 {
  display: inline-block;
  vertical-align: sub;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  color: currentColor;
}

<div>
  <button
    aria-label="Test"
    aria-pressed="false"
    class="pounce-1"
    type="button"
  >
    <svg
      class="pounce-0"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 13.1429H13.1429V20H10.8571V13.1429H4V10.8571H10.8571V4H13.1429V10.8571H20V13.1429Z"
      />
    </svg>
  </button>
</div>
`;

exports[`IconButton renders \`small\` size correctly 1`] = `
.pounce-1 {
  outline: none;
  line-height: 1;
  border-radius: 4px;
  height: 24px;
  width: 24px;
  -webkit-transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border: 1px solid;
  border-color: #0081C5;
  background-color: #0081C5;
  cursor: pointer;
  color: #F6F6F6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.pounce-1:hover {
  background-color: #0a8bcf;
  border-color: #0a8bcf;
}

.pounce-1:focus {
  background-color: #0a8bcf;
  border-color: #55d6ff;
}

.pounce-1:active,
.pounce-1[data-active=true] {
  background-color: #0077bb;
  border-color: #0077bb;
}

.pounce-0 {
  display: inline-block;
  vertical-align: sub;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 16px;
  height: 16px;
  color: currentColor;
}

<div>
  <button
    aria-label="Test"
    aria-pressed="false"
    class="pounce-1"
    type="button"
  >
    <svg
      class="pounce-0"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 13.1429H13.1429V20H10.8571V13.1429H4V10.8571H10.8571V4H13.1429V10.8571H20V13.1429Z"
      />
    </svg>
  </button>
</div>
`;

exports[`IconButton renders \`square\` variantBorderStyle correctly 1`] = `
.pounce-1 {
  outline: none;
  line-height: 1;
  border-radius: 4px;
  height: 46px;
  width: 46px;
  -webkit-transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border: 1px solid;
  border-color: #0081C5;
  background-color: #0081C5;
  cursor: pointer;
  color: #F6F6F6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.pounce-1:hover {
  background-color: #0a8bcf;
  border-color: #0a8bcf;
}

.pounce-1:focus {
  background-color: #0a8bcf;
  border-color: #55d6ff;
}

.pounce-1:active,
.pounce-1[data-active=true] {
  background-color: #0077bb;
  border-color: #0077bb;
}

.pounce-0 {
  display: inline-block;
  vertical-align: sub;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  color: currentColor;
}

<div>
  <button
    aria-label="Test"
    aria-pressed="false"
    class="pounce-1"
    type="button"
  >
    <svg
      class="pounce-0"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 13.1429H13.1429V20H10.8571V13.1429H4V10.8571H10.8571V4H13.1429V10.8571H20V13.1429Z"
      />
    </svg>
  </button>
</div>
`;

exports[`IconButton renders \`variantColor\`  and \`variants\` correctly 1`] = `
.pounce-0 {
  display: inline-block;
  vertical-align: sub;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  color: currentColor;
}

.pounce-1 {
  outline: none;
  line-height: 1;
  border-radius: 4px;
  height: 46px;
  width: 46px;
  -webkit-transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border: 1px solid;
  border-color: #09A084;
  background-color: #09A084;
  cursor: pointer;
  color: #F6F6F6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.pounce-1:hover {
  background-color: #13aa8e;
  border-color: #13aa8e;
}

.pounce-1:focus {
  background-color: #13aa8e;
  border-color: #5ef5d9;
}

.pounce-1:active,
.pounce-1[data-active=true] {
  background-color: #00967a;
  border-color: #00967a;
}

.pounce-3 {
  outline: none;
  line-height: 1;
  border-radius: 4px;
  height: 46px;
  width: 46px;
  -webkit-transition: background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border: 1px solid;
  border-color: transparent;
  background-color: transparent;
  cursor: pointer;
  color: #F6F6F6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.pounce-3:hover {
  background-color: rgba(135,55,75,0.3);
}

.pounce-3:focus {
  background-color: rgba(135,55,75,0.3);
}

.pounce-3:active,
.pounce-3[data-active=true] {
  background-color: #87374B;
}

.pounce-5 {
  outline: none;
  line-height: 1;
  border-radius: 4px;
  height: 46px;
  width: 46px;
  -webkit-transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border: 1px solid;
  border-color: #141C2A;
  background-color: #141C2A;
  cursor: pointer;
  color: #F6F6F6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.pounce-5:hover {
  background-color: #1e2634;
  border-color: #1e2634;
}

.pounce-5:focus {
  background-color: #1e2634;
  border-color: #69717f;
}

.pounce-5:active,
.pounce-5[data-active=true] {
  background-color: #0a1220;
  border-color: #0a1220;
}

<div>
  <button
    aria-label="Test"
    aria-pressed="false"
    class="pounce-1"
    type="button"
  >
    <svg
      class="pounce-0"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 13.1429H13.1429V20H10.8571V13.1429H4V10.8571H10.8571V4H13.1429V10.8571H20V13.1429Z"
      />
    </svg>
  </button>
  <button
    aria-label="Test red"
    aria-pressed="false"
    class="pounce-3"
    type="button"
  >
    <svg
      class="pounce-0"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M9.13 21.156l.446-1.308c-1.07-.347-1.624-1.293-1.661-2.84v-1.975c0-1.404-.535-2.39-1.605-2.96 1.07-.563 1.605-1.54 1.605-2.931V7.138c.037-1.547.59-2.493 1.66-2.84L9.13 3c-1.088.31-1.912.807-2.472 1.494-.56.686-.849 1.586-.867 2.7V9.16c0 1.398-.597 2.097-1.791 2.097v1.651c1.194 0 1.79.696 1.79 2.088v1.976c.044 2.177 1.157 3.571 3.34 4.184zm5.796 0c1.12-.316 1.956-.827 2.51-1.536.553-.708.83-1.64.83-2.797v-1.827c0-1.392.6-2.088 1.8-2.088v-1.651c-1.2 0-1.8-.696-1.8-2.088V7.314c0-1.144-.276-2.07-.826-2.779-.55-.708-1.389-1.22-2.514-1.535l-.446 1.299c1.089.352 1.646 1.317 1.67 2.894v1.92c0 1.448.563 2.438 1.689 2.97-1.126.531-1.689 1.512-1.689 2.94v1.93c-.024 1.577-.581 2.542-1.67 2.895l.446 1.308z"
      />
    </svg>
  </button>
  <button
    aria-label="Test timer"
    aria-pressed="false"
    class="pounce-5"
    type="button"
  >
    <svg
      class="pounce-0"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 17C11 17.55 11.45 18 12 18C12.55 18 13 17.55 13 17C13 16.45 12.55 16 12 16C11.45 16 11 16.45 11 17ZM11 3V7H13V5.08C16.39 5.57 19 8.47 19 12C19 15.87 15.87 19 12 19C8.13 19 5 15.87 5 12C5 10.32 5.59 8.78 6.58 7.58L12 13L13.41 11.59L6.61 4.79V4.81C4.42 6.45 3 9.05 3 12C3 16.97 7.02 21 12 21C16.97 21 21 16.97 21 12C21 7.03 16.97 3 12 3H11ZM18 12C18 11.45 17.55 11 17 11C16.45 11 16 11.45 16 12C16 12.55 16.45 13 17 13C17.55 13 18 12.55 18 12ZM6 12C6 12.55 6.45 13 7 13C7.55 13 8 12.55 8 12C8 11.45 7.55 11 7 11C6.45 11 6 11.45 6 12Z"
      />
    </svg>
  </button>
</div>
`;
